index.html

```diff
@@ -833,23 +839,70 @@ <h3>Extensions to the <code>Performance</code> Interface</h3>
 'PERFORMANCE-TIMELINE-2#dfn-performance-entry-buffer'>performance
 entry buffer</a>.</li>
 <li>Increase <a>resource timing buffer current size</a> by 1.</li>
+<li>Return.</li>
 </ol>
 </li>
-<li>If <a>resource timing buffer current size</a> is greater than
-or equal to <a>resource timing buffer size limit</a> and the
-<a>resource timing buffer full flag</a> is false, run the following
+<li>If <a>resource timing buffer full flag</a> is false, run the following
 substeps:
-<ol style='list-style-type: lower-latin;' data-link-for=
-"Performance">
-<li>Set the <a>resource timing buffer full flag</a> to true.</li>
-<li><a data-cite="DOM#concept-event-fire">Fire an event</a> named
-<code>resourcetimingbufferfull</code> at the <a data-cite=
-"HR-TIME-2#idl-def-performance">Performance</a> object, with its
-<code>bubbles</code> attribute initialized to true.</li>
+<ol style="list-style-type: lower-latin;">
+<li>Set <a>resource timing buffer full flag</a> to true.</li>
+<li>Queue a task to run <a>fire a buffer full event</a>.</li>
+</ol>
+<li>Add <i>new entry</i> to the <a>secondary entry buffer</a>.</li>
+</ol>
+
+<p>To <dfn>copy secondary buffer</dfn>, run the following steps:</p>
```

Comment:
I think this algorithm needs to check that the performance entry buffer size limit is never exceeded.

Reply:
To clarify, the reason this is important is that developers could query entries in the buffer full event, and this algorithm is saying that they will get all of them even though they're not supposed to be in the performance timeline yet.

Reply:
so instead of sync firing the event when the primary exceeds, we'd fire it until the secondary is empty? I guess that can work